### ISSUE TITLE
Windows .manifest file included in onefile bootloader specifies not-yet-existing dependencies

### ISSUE THREAD

**Issue by codewarrior0** [pyinstaller/pyinstaller#1180]
I'm trying to figure out why the MSVCR90.dll and others aren't being included in the onefile archive. This is probably because I need to add them to the binaries TOC. But first I have a quick question to ask.

Why, if the bootloader `run.exe` does not itself depend on those .dlls, is a .manifest being embedded into it that specifies a dependency on them? This apparently causes a "side-by-side configuration is incorrect" error for the bootloader on a system where the redist package is not installed, before it even gets to unpacking the archive.

**Comment by codewarrior0** [comment]
I've found out that MCVCR90.dll is in fact being included into the archive.

I've also found out that `run.exe` does indirectly depend on those .dlls, but only after it has unpacked the archive and loaded python27.dll and friends. It seems to me that it is incorrect to embed a manifest into the bootloader when those files won't exist until after the bootloader has unpacked itself. Yet, a manifest will be embedded into it when UAC admin rights, etc. are requested. It is also correct to embed a manifest for other dependencies that weren't collected into the archive.

I've found a [mailing list post](http://comments.gmane.org/gmane.comp.python.pyinstaller/6852) where the author had to go in and edit the .manifest file after it was generated by the Analysis step, and remove the MSVCR dependencies before his app would run on a machine without them. I suspect the fix is to NOT add dependencies to the .manifest file when they are ones that will be packed into the onefile archive.

**Comment by codewarrior0** [comment]
Possibly related to #772

**Comment by codewarrior0** [comment]
Whoops, the mailing list post isn't the one I thought I linked. I can't find the one I was looking at now. But that one does describe a different issue: having different .pyd or .dll files included, whose manifests each have dependencies on different versions of the assembly. The .dll files are included only once, instead of including the different assemblies as complete items... but that's a separate issue.

**Comment by codewarrior0** [comment]
All right, I think I understand this problem now. In onefile mode, the .exe manifest specifies dependencies on the files it is about to extract, but since those files don't exist yet, the windows exe loader refuses to even start the .exe so it can extract its own dependencies. 

I can't just remove the manifest, because then it can't link up with the .dlls it needs after extracting them. The fix I found is to modify the manifest just before inserting it into the .exe, and set all of the dependent assemblies to optional. Will test it out a few more times, then send a PR.

**Comment by htgoebel** [comment]
Merged  #1182 fixes this issue.
